### PR TITLE
Ensure that searcher.getReader() is used when using Lucene test searcher

### LINKED ISSUE #98925: `org.elasticsearch.search.internal.ContextIndexSearcherTests.testIntersectScorerAndRoleBits` failing in main
### CI Link

https://gradle-enterprise.elastic.co/s/n6mmfb3uwtq7q

### Repro line

./gradlew ':server:test' --tests "org.elasticsearch.search.internal.ContextIndexSearcherTests.testIntersectScorerAndRoleBits" \   -Dtests.seed=E99DBC08AEFDABC1 \   -Dtests.locale=und \   -Dtests.timezone=America/Santarem \   -Druntime.java=20

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

10:13:31 org.elasticsearch.search.internal.ContextIndexSearcherTests > testIntersectScorerAndRoleBits FAILED
10:13:31     java.lang.AssertionError: The top-reader used to create Weight is not the same as the current reader's top-reader (org.apache.lucene.index.CompositeReaderContext@4afc5040
10:13:31         at __randomizedtesting.SeedInfo.seed([E99DBC08AEFDABC1:41ACDD81C7F3BEAA]:0)
10:13:31         at org.apache.lucene.search.TermQuery$TermWeight.scorer(TermQuery.java:105)
10:13:31         at org.apache.lucene.search.Weight.scorerSupplier(Weight.java:136)
10:13:31         at org.apache.lucene.search.LRUQueryCache$CachingWrapperWeight.scorerSupplier(LRUQueryCache.java:759)
10:13:31         at org.apache.lucene.tests.search.AssertingWeight.scorerSupplier(AssertingWeight.java:80)
10:13:31         at org.apache.lucene.search.ConstantScoreQuery$1.scorerSupplier(ConstantScoreQuery.java:153)
10:13:31         at org.apache.lucene.tests.search.AssertingWeight.scorerSupplier(AssertingWeight.java:80)
10:13:31         at org.apache.lucene.tests.search.AssertingWeight.scorer(AssertingWeight.java:66)
10:13:31         at org.elasticsearch.search.internal.ContextIndexSearcherTests.testIntersectScorerAndRoleBits(ContextIndexSearcherTests.java:157)

### PR DESCRIPTION
**Pull request by benwtrent**
Related to: https://github.com/elastic/elasticsearch/pull/98110

Lucene test searcher will randomly wrap the passed reader. That means any usages that mix the two (the searcher and the previously reader), could cause tests to fail.

Some of these locations maybe don't strictly need the searcher.getReader, but I just wanted to make sure we didn't have flaky tests.

closes: https://github.com/elastic/elasticsearch/issues/98925

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/index/search/nested/NestedSortingTests.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ public void testDuel() throws Exception {
         MultiValueMode sortMode = randomFrom(Arrays.asList(MultiValueMode.MIN, MultiValueMode.MAX));
         DirectoryReader reader = DirectoryReader.open(writer);
         reader = ElasticsearchDirectoryReader.wrap(reader, new ShardId(indexService.index(), 0));
-        IndexSearcher searcher = newSearcher(reader);
+        IndexSearcher searcher = new IndexSearcher(reader);
```
Comment (benwtrent) [inline server/src/test/java/org/elasticsearch/index/search/nested/NestedSortingTests.java:102]: Dear reviewer, this fails randomly due to the reader getting randomly wrapped internally by `newSearcher`. We expect this type of reader and it cannot be wrapped. 

```
org.elasticsearch.index.search.nested.NestedSortingTests > testMultiLevelNestedSorting {seed=[4A3E25BDAA026372:B462EF24EC743438]} FAILED
    java.lang.IllegalStateException: Null shardId. If you got here from a test, you need to wrap the directory reader. see for example AggregatorTestCase#wrapInMockESDirectoryReader.  If you got here in production, please file a bug.
        at __randomizedtesting.SeedInfo.seed([4A3E25BDAA026372:B462EF24EC743438]:0)
        at org.elasticsearch.index.cache.bitset.BitsetFilterCache.getAndLoadIfNotPresent(BitsetFilterCache.java:149)
        at org.elasticsearch.index.cache.bitset.BitsetFilterCache$QueryWrapperBitSetProducer.getBitSet(BitsetFilterCache.java:224)
        at org.elasticsearch.index.fielddata.IndexFieldData$XFieldComparatorSource$Nested.rootDocs(IndexFieldData.java:141)
        at org.elasticsearch.index.fielddata.fieldcomparator.LongValuesComparatorSource.getNumericDocValues(LongValuesComparatorSource.java:90)
        at org.elasticsearch.index.fielddata.fieldcomparator.LongValuesComparatorSource$1$1.getNumericDocValues(LongValuesComparatorSource.java:109)
        at org.apache.lucene.search.comparators.NumericComparator$NumericLeafComparator.<init>(NumericComparator.java:106)
        at org.apache.lucene.search.comparators.LongComparator$LongLeafComparator.<init>(LongComparator.java:65)
        at org.elasticsearch.index.fielddata.fieldcomparator.LongValuesComparatorSource$1$1.<init>(LongValuesComparatorSource.java:106)
        at org.elasticsearch.index.fielddata.fieldcomparator.LongValuesComparatorSource$1.getLeafComparator(LongValuesComparatorSource.java:106)
        at org.apache.lucene.search.FieldValueHitQueue.getComparators(FieldValueHitQueue.java:176)
        at org.apache.lucene.search.TopFieldCollector$TopFieldLeafCollector.<init>(TopFieldCollector.java:64)
        at org.apache.lucene.search.TopFieldCollector$SimpleFieldCollector$1.<init>(TopFieldCollector.java:198)
        at org.apache.lucene.search.TopFieldCollector$SimpleFieldCollector.getLeafCollector(TopFieldCollector.java:198)
        at org.apache.lucene.search.FilterCollector.getLeafCollector(FilterCollector.java:38)
        at org.apache.lucene.tests.search.AssertingCollector.getLeafCollector(AssertingCollector.java:52)
        at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:762)
        at org.apache.lucene.tests.search.AssertingIndexSearcher.search(AssertingIndexSearcher.java:78)
        at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:694)
        at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:688)
        at org.apache.lucene.search.IndexSearcher.searchAfter(IndexSearcher.java:668)
        at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:585)
        at org.elasticsearch.index.search.nested.NestedSortingTests.search(NestedSortingTests.java:823)
        at org.elasticsearch.index.search.nested.NestedSortingTests.testMultiLevelNestedSorting(NestedSortingTests.java:621)
```

File: server/src/test/java/org/elasticsearch/index/search/nested/NestedSortingTests.java [hunk 2]
```diff
@@ -99,7 +99,7 @@ public void testDuel() throws Exception {
         MultiValueMode sortMode = randomFrom(Arrays.asList(MultiValueMode.MIN, MultiValueMode.MAX));
         DirectoryReader reader = DirectoryReader.open(writer);
         reader = ElasticsearchDirectoryReader.wrap(reader, new ShardId(indexService.index(), 0));
-        IndexSearcher searcher = newSearcher(reader);
+        IndexSearcher searcher = newSearcher(reader, false);
```
Comment (benwtrent) [inline server/src/test/java/org/elasticsearch/index/search/nested/NestedSortingTests.java:102]: This test fails when the reader is wrapped. So making it not `maybeWrap` :)

**Review by iverase — CHANGES_REQUESTED** [review]
oh, just before reviewing this PR I opened https://github.com/elastic/elasticsearch/pull/98960 which fixes the `FiltersAggregatorTests` in a different way. I feel the approach of my PR is more inline with the needs for that test. The reason is that we are moving on supporting concurrency and the lucene index searcher behavior is very different to the Elasticsearch one so we should be using the later.

I will ask you to revert the changes to that test (I will push the above Pr as soon as it is reviewed), otherwise looks good.

**Comment by benwtrent** [comment]
@iverase reverted by filteraggs change and merged in main.

**Review by iverase — APPROVED** [review]
LGTM

**Pull request merged**